docs/conf/scalar-loki-stack-custom-values.yaml

```diff
@@ -0,0 +1,50 @@
+promtail:
+  config:
+    snippets:
+      scrapeConfigs: |
+        - job_name: scalardl-ledger
+          pipeline_stages:
+            - docker: {}
+          kubernetes_sd_configs:
+            - role: pod
+          relabel_configs:
+            - source_labels:
+                - __meta_kubernetes_pod_node_name
+              target_label: __host__
+            - action: replace
+              source_labels:
+                - __meta_kubernetes_pod_name
+              target_label: pod
+            - action: keep
+              regex: scalardl-ledger-(.+)
```

Comment:
I think this configuration will drop Scalar product logs, when users set arbitrary `release name` of helm of Ledger and Auditor.
So, I think it is better to change regex configuration to detect string `scalardl` like `(.*)scalardl-(.+)`.

I will describe the details as follows.

In the following two documents, we set `release name` to `scalar-ledger` and `scalar-auditor` as sample.
* [Getting Started with Helm Charts (Scalar DL Ledger)](https://github.com/scalar-labs/helm-charts/blob/1864a73979519f483f671616ba7d8b8142a4a149/docs/getting-started-scalardl-ledger.md)
* [Getting Started with Helm Charts (Scalar DL Auditor)](https://github.com/scalar-labs/helm-charts/blob/1864a73979519f483f671616ba7d8b8142a4a149/docs/getting-started-scalardl-auditor.md)

So, if we according to the document completely, the promtail get scalar product logs.
In other words, if we according to the document completely, the pod name `scalardl-ledger-ledger-xxxx` and `scalardl-ledger-ledger-envoy-xxxx` matches the regex `scalardl-ledger-(.+)`.

However, when we set `release name` to some arbitrary name, the pod name will be the following.

```
$ kubectl get pod
NAME                                                    READY   STATUS    RESTARTS   AGE
my-project-aaa-scalardl-envoy-557bb44845-j4mbp          1/1     Running   0          4m25s
my-project-aaa-scalardl-ledger-5cdb8dddc9-8p9fz         1/1     Running   0          4m25s
my-project-bbb-scalardl-audit-auditor-d5466c8d5-5kgm9   1/1     Running   0          2m36s
my-project-bbb-scalardl-audit-envoy-5b7f85fff5-fmtc6    1/1     Running   0          2m36s
```
* I deployed Ledger and Auditor by `helm install my-project-aaa ...` and `helm install my-project-bbb ...` command.

In this case, the above pod name does not match the regex `scalardl-ledger-(.+)`.
So, the promtail doesn't send scalar product logs to Loki Server, and we cannot see the logs in Grafana Dashboard.

* It seems that if the pod (label) name does not match, promtail drops logs.
  https://grafana.com/docs/loki/latest/clients/promtail/configuration/#relabel_configs

Also, the following message output periodically in the promtail log, in spite of there are log files in the Kubernetes node.

* message
  ```
  level=warn ts=2022-05-09T07:31:28.760478726Z caller=logging.go:72 msg="GET /ready (500) 71.174µs Response: \"Not ready: Unable to find any logs to tail. Please verify permissions, volumes, scrape_config, etc.\\n\" ws: false; Accept: */*; Connection: close; User-Agent: kube-probe/1.23; "
  ```

* log files exist
  ```
  docker@minikube:~$ ls -l /var/log/pods | grep scalardl
  drwxr-xr-x 3 root root 4096 May  9 06:50 default_my-project-aaa-scalardl-envoy-557bb44845-j4mbp_b701d0f1-5078-46d2-b9de-36361c5b68b2
  drwxr-xr-x 3 root root 4096 May  9 06:50 default_my-project-aaa-scalardl-ledger-5cdb8dddc9-8p9fz_aaf1501b-315a-49f3-be8b-db7f2c3c38e3
  drwxr-xr-x 3 root root 4096 May  9 06:52 default_my-project-bbb-scalardl-audit-auditor-d5466c8d5-5kgm9_c42698bf-02f9-464b-bb56-9edd02b87360
  drwxr-xr-x 3 root root 4096 May  9 06:52 default_my-project-bbb-scalardl-audit-envoy-5b7f85fff5-fmtc6_07870969-69f2-4d2b-8e1b-1234111f3c90
  ```

So, I changed the configuration of promtail as follows.

```yaml
promtail:
  config:
    snippets:
      scrapeConfigs: |
        - job_name: scalardl

...

            - action: keep
              regex: (.*)scalardl-(.+)
              source_labels:
                - pod
```

In the result, promtail send logs to Loki Server correctly, and I can see the logs in the Grafana Dashboard.
Also, promtail output the following logs. It seems that the promtail detects log files of Scalar product.

* promtail log
  ```
  level=info ts=2022-05-09T07:38:35.701011981Z caller=filetargetmanager.go:255 msg="Adding target" key="/var/log/pods/*ab99358f-21fb-41db-9b33-521513438352/scalardl-ledger/*.log:{pod=\"my-project-aaa-scalardl-ledger-5cdb8dddc9-hpfrg\"}" ts=2022-05-09T07:38:35.701809292Z caller=log.go:168 
  level=info msg="Seeked /var/log/pods/default_my-project-aaa-scalardl-ledger-5cdb8dddc9-hpfrg_ab99358f-21fb-41db-9b33-521513438352/scalardl-ledger/0.log - &{Offset:0 Whence:0}"
  level=info ts=2022-05-09T07:38:35.701889928Z caller=tailer.go:126 component=tailer msg="tail routine: started" path=/var/log/pods/default_my-project-aaa-scalardl-ledger-5cdb8dddc9-hpfrg_ab99358f-21fb-41db-9b33-521513438352/scalardl-ledger/0.log
  ```

So, I think that it is better to set the regex configuration to match string `scalardl` (For example, `(.*)scalardl-(.+)`).
Basically, in the Helm Charts of Ledger and Auditor, the string `scalardl` is included in the pod name.

* [Ledger](https://github.com/scalar-labs/helm-charts/blob/scalardl-4.2.1/charts/scalardl/templates/_helpers.tpl#L14-L22)
* [Auditor](https://github.com/scalar-labs/helm-charts/blob/scalardl-audit-2.2.1/charts/scalardl-audit/templates/_helpers.tpl#L14-L22)
  * If users set `nameOverride` or `fullNameOverride`, we cannot detect logs (pod name) by `(.*)scalardl-(.+)`. However, I think that there is no configuration that covers all pattern of `release name`. So, I think it is enough covering basically use case in the sample custom value file.

So, I think that we can detect Scalar DL logs by one configuration `(.*)scalardl-(.+)` only.

Reply:
Thank you for the advice.
This is more generic!
I revised it in https://github.com/scalar-labs/helm-charts/pull/87/commits/4dae167456835cacc6da4ea24bada3760f0738d2